src/Language/PureScript/CoreImp/Optimizer/Unused.hs

```diff
@@ -20,8 +20,11 @@ removeCodeAfterReturnStatements :: AST -> AST
 removeCodeAfterReturnStatements = everywhere (removeFromBlock go)
   where
   go :: [AST] -> [AST]
-  go jss | not (any isReturn jss) = jss
-         | otherwise = let (body, ret : _) = break isReturn jss in body ++ [ret]
+  go jss =
+    case break isReturn jss of
```

Comment:
This has the exact same behaviour as before; `not (any isReturn xs)` is just as expensive to compute as `break isReturn xs`, so we might as well do the latter since that also gives us access to the information we need here in case there is a return statement.